libipl/libipl.C

```diff
@@ -359,3 +362,43 @@ void ipl_error_callback(const ipl_error_info& error)
 		return;
 	g_ipl_error_callback_fn(error);
 }
+
+void ipl_process_fapi_error(const fapi2::ReturnCode& fapirc,
+		struct pdbg_target *target, bool deconfig)
+{
+	if(fapirc == fapi2::FAPI2_RC_SUCCESS) {
+		ipl_error_callback(IPL_ERR_OK);
+	} 
+	else if(fapirc.getCreator() == fapi2::ReturnCode::CREATOR_PLAT) {
+		CDG_Target cdgTarget;
+		cdgTarget.callout_priority =
+		fapi2::plat_CalloutPriority_tostring(fapi2::CalloutPriorities::MEDIUM);
+
+		ATTR_PHYS_BIN_PATH_Type physBinPath;
+		uint32_t binPathElemCount = dtAttr::fapi2::ATTR_PHYS_BIN_PATH_ElementCount;
+		if (!pdbg_target_get_attribute(target, "ATTR_PHYS_BIN_PATH",
+			std::stoi(dtAttr::fapi2::ATTR_PHYS_BIN_PATH_Spec),
+				binPathElemCount, physBinPath)) {
+			ipl_log(IPL_ERROR, "Failed to read ATTR_PHYS_BIN_PATH for proc %s\n",
+				pdbg_target_path(target));
+			ipl_error_callback(IPL_ERR_ATTR_READ_FAIL);
+		}
+		else {
+			std::copy(physBinPath, physBinPath+binPathElemCount,
+			std::back_inserter(cdgTarget.target_entity_path));
+			cdgTarget.deconfigure = deconfig;
+
+			FFDC* ffdc = new FFDC();;
```

Comment:
Can you add a comment about `free` I think it will be taken care of by the `ipl_error_info` destructor, right?

since you are using `new`, are you having any cleanup in the FFDC structure? I think,  ipl_error_info using `free` so that won't call the destructor of FFDC.

Reply:
I think free in ~ipl_error_info() is not required.
Since this is a structure used in callback, we can always pass a pointer to local variable safely, since its scope will never end before the callback returns.

Reply:
good catch, damn how whould we fix this?

Reply:
rajees, i had reverted back freeing buffer in destructor as jayanth meant it to be used to allocate some buffer and free it after it goes out of scope.

Reply:
>I think free in ~ipl_error_info() is not required.
Since this is a structure used in callback,

that `ipl_error_info` is common and Jayanth added that for some other use case.

> good catch, damn how whould we fix this?
Yeah, it is a tricky one :) I need to think about it.

Reply:
now decided to call delete in the desctrucor instead of free, jayanth proposed this solution